spec/models/teacher_spec.rb

```diff
@@ -8,4 +8,16 @@
     it { is_expected.to validate_presence_of(:email) }
     it { is_expected.to validate_presence_of(:password) }
   end
+
+  describe 'email validation fail' do
+    it 'for unique email' do
+      teacher_1 = create(:teacher, attributes_for(:teacher))
+      teacher_2 = build(:teacher, attributes_for(:teacher, email: teacher_1.email))
+      expect(teacher_2).not_to be_valid
```

Comment:
`shoulda-matchers` gem actually provides you with [matcher for validating uniqueness](https://github.com/thoughtbot/shoulda-matchers/blob/main/lib/shoulda/matchers/active_record/validate_uniqueness_of_matcher.rb)

This is one of the harder matchers to use as you have to create the record first to validate its uniqueness. One of good examples is [here](https://github.com/thoughtbot/shoulda-matchers/blob/main/lib/shoulda/matchers/active_record/validate_uniqueness_of_matcher.rb#L88)

Also keep in mind we agreed to prefer `is_expected.to` over `should` :)

Reply:
Check :)